riscv/processor.cc

```diff
@@ -462,7 +462,12 @@ void state_t::reset(processor_t* const proc, reg_t max_isa)
   if (proc->extension_enabled_const(EXT_SSTC)) {
     stimecmp = std::make_shared<stimecmp_csr_t>(proc, CSR_STIMECMP, MIP_STIP);
     csrmap[CSR_VSTIMECMP] = vstimecmp = std::make_shared<stimecmp_csr_t>(proc, CSR_VSTIMECMP, MIP_VSTIP);
-    csrmap[CSR_STIMECMP] = std::make_shared<virtualized_stimecmp_csr_t>(proc, stimecmp, vstimecmp);
+    auto virtualized_stimecmp = std::make_shared<virtualized_stimecmp_csr_t>(proc, stimecmp, vstimecmp);
+    csrmap[CSR_STIMECMP] = virtualized_stimecmp;
```

Comment:
You need to use `rv32_low_csr_t` for the lower half; otherwise CSRW will erroneously zero the upper 32 bits. See above for `CSR_HENVCFG` for an example.